src/main/java/com/wavefront/sdk/direct_ingestion/WavefrontDirectIngestionClient.java

```diff
@@ -34,32 +39,93 @@
  *
  * @author Vikram Raman (vikram@wavefront.com)
  */
-public class WavefrontDirectIngestionClient extends AbstractDirectConnectionHandler
-    implements WavefrontMetricSender, WavefrontHistogramSender, WavefrontTracingSpanSender {
+public class WavefrontDirectIngestionClient implements WavefrontMetricSender,
+    WavefrontHistogramSender, WavefrontTracingSpanSender, BufferFlusher, Runnable, Closeable {
 
   private static final String DEFAULT_SOURCE = "wavefrontDirectSender";
   private static final Logger LOGGER = Logger.getLogger(
       WavefrontDirectIngestionClient.class.getCanonicalName());
-  private static final int MAX_QUEUE_SIZE = 50000;
-  private static final int BATCH_SIZE = 10000;
-
-  private final LinkedBlockingQueue<String> metricsBuffer =
-      new LinkedBlockingQueue<>(MAX_QUEUE_SIZE);
-  private final LinkedBlockingQueue<String> histogramsBuffer =
-      new LinkedBlockingQueue<>(MAX_QUEUE_SIZE);
-  private final LinkedBlockingQueue<String> tracingSpansBuffer =
-      new LinkedBlockingQueue<>(MAX_QUEUE_SIZE);
 
   private final AtomicInteger failures = new AtomicInteger();
+  private final int batchSize;
+  private final LinkedBlockingQueue<String> metricsBuffer;
+  private final LinkedBlockingQueue<String> histogramsBuffer;
+  private final LinkedBlockingQueue<String> tracingSpansBuffer;
+  private final DataIngesterAPI directService;
+  private final ScheduledExecutorService scheduler;
+
+  public static class Builder {
+    // Required parameters
+    private final String server;
+    private final String token;
+
+    // Optional parameters
+    private int maxQueueSize = 50000;
+    private int batchSize = 10000;
+    private int flushIntervalSeconds = 1;
+
+    /**
+     * Create a new WavefrontDirectIngestionClient.Builder
+     *
+     * @param server A Wavefront server URL of the form "https://clusterName.wavefront.com"
+     * @param token A valid API token with direct ingestion permissions
+     */
+    public Builder(String server, String token) {
+      this.server = server;
+      this.token = token;
+    }
+
+    /**
+     * Set max queue size of in-memory buffer. Needs to be flushed if full.
+     *
+     * @param maxQueueSize Max queue size of in-memory buffer
+     * @return WavefrontDirectIngestionClient.Builder instance
```

Comment:
`@return {@code this}` for the methods that return `Builder`.

Reply:
We should probably make these changes to the WavefrontProxyClient as well to be consistent.